### ISSUE TITLE
terrible performance on linux

### ISSUE THREAD

**Issue by djdunn** [CompletelyFairGames/dwarfcorp#690]
haven't tried to play for a few months but now i get <1 fps and after spending a few minutes to slowly move the mouse to the pause button and pause, then the fps speeds up to where it should be.

even on lowest settings

some system info
https://paste.pound-python.org/show/xnkexqqFYICzxgHVF3VL/

glxinfo 
https://paste.pound-python.org/show/lyX9cKwrC6rdEDdiTLJt/

**Comment by mklingen** [comment]
This must be due to the caching we implemented for designation drawing. Perhaps on linux the dynamic vertex buffer isn't set up correctly.

**Comment by mklingen** [comment]
Ok thank you that's valuable information. It's update, not draw. Unfortunately that data doesn't show us where most of those 50ms are coming from. For the record I can run at a solid 60 FPS all the time on Ubuntu 14.04 with a GTX 860m. The log spam might be a hint. It could be slowing down because of that warning getting spammed to the log.

**Comment by mklingen** [comment]
Will try to reproduce/investigate the issue. Thanks for your data.

**Comment by com1clyf3** [comment]
@djdunn Hey! So, in our newer builds, performance debug no longer looks like this. So I'm guess you either need to update your build or we haven't updated the Linux build recently. @mklingen We push new linux and mac builds when we push steam update builds, right? Or no?

**Comment by mklingen** [comment]
Yes, we do. The last release was before that change, however. We will be pushing a new version likely this weekend.

**Comment by com1clyf3** [comment]
Oh right. My bad.

**Comment by mklingen** [comment]
it does? Yay!

**Comment by djdunn** [comment]
Yeah it's not the instant 1/10 fps but nice for awhile till it slows down considerably, will post more in the other thread when I can get performance overlay working

**Comment by mklingen** [comment]
performance overlay currently doesn't work on linux/mac.
